### PR TITLE
Christian merchant

### PR DESCRIPTION
**Pull request by cvalesares**
add both merchant invoice show page and admin merchant show page

### CODE REVIEW

File: app/models/item.rb [hunk 1]
```diff
@@ -1,4 +1,9 @@
 class Item < ApplicationRecord
   has_many :invoice_items
   belongs_to :merchant
+
+  def self.find_item(id)
+    find(id)
+  end
```
Comment (samueldevine) [inline app/models/item.rb]: this is an existing ActiveRecord method! `Item.find(id)` should work - no need to redefine it here

File: app/views/merchant_invoices/show.html.erb [hunk 2]
```diff
@@ -1 +1,8 @@
-<h1>This is MerchantInvoice Show Page</h1>
+<h1> Invoice #<%= @invoice.id %> </h1>
+
+<% @invoice.invoice_items.each do |invoice_item| %>
+  <%= Item.find_item(invoice_item.item_id).name %><br>
+  Price: <%= Item.find_item(invoice_item.item_id).unit_price %><br>
```
Comment (samueldevine) [inline app/views/merchant_invoices/show.html.erb]: mentioned in the Item model, you can just use the existing AR find method: `Item.find(invoice_item.item_id).name`

File: spec/features/merchant_admin/show_spec.rb [hunk 3]
```diff
@@ -0,0 +1,20 @@
+require 'rails_helper'
+
+RSpec.describe "Merchant Admin Show" do
+  it 'can show merchant name' do
+    merchant1 = Merchant.create!(name: "Jimmy Pesto")
+    merchant2 = Merchant.create!(name: "Linda Belcher")
```
Comment (samueldevine) [inline spec/features/merchant_admin/show_spec.rb:6]: lol'd @ jimmy pesto

File: spec/features/merchant_invoice/show_spec.rb [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+
+RSpec.describe "Merchant Invoice Show" do
+  it 'shows all invoice items' do
+    merchant1 = Merchant.create!(name: "Bob's Burgers")
+    customer1 = Customer.create!(first_name: "Teddy", last_name: "Lastname")
+    invoice1 = Invoice.create!(customer_id: customer1.id, status: 1)
+    item1 = Item.create!(name: "Burger", description: "Burger with fries", unit_price: 3, merchant_id: merchant1.id)
+    item2 = Item.create!(name: "Shake", description: "Shake without fries", unit_price: 2, merchant_id: merchant1.id)
+    invoiceitem1 = InvoiceItem.create!(item_id: item1.id, invoice_id: invoice1.id, quantity: 3, unit_price: item1.unit_price, status: 0)
+    invoiceitem2 = InvoiceItem.create!(item_id: item2.id, invoice_id: invoice1.id, quantity: 2, unit_price: item2.unit_price, status: 0)
```
Comment (samueldevine) [inline spec/features/merchant_invoice/show_spec.rb]: it doesn't look like you're using item2 and invoiceitem2 here?

File: spec/models/item_spec.rb [hunk 5]
```diff
@@ -5,4 +5,13 @@
     it {should have_many :invoice_items}
     it {should belong_to :merchant}
   end
+
+  describe "::find_item" do
```
Comment (samueldevine) [inline spec/models/item_spec.rb]: no need for this test if you get rid of the method I mentioned above

**Review by samueldevine — DISMISSED** [review]
nice job! I noticed that you are defining a find_item method that you don't need - it's already an existing activerecord method :) 
you should be able to delete that method and the model test for it, and then there's a light refactor to use the existing AR method in the view. once that's done, all the rest looks ready to merge!

File: spec/features/merchant_invoice/show_spec.rb (hunk 4 above)
Comment (cvalesares) [reply spec/features/merchant_invoice/show_spec.rb]: good catch! darn you copy and paste

**Pull request merged**